### PR TITLE
Fix development environments

### PR DESCRIPTION
**Pull request by feluelle**
# Description

- fix local.mk to respect the correct directory
- fix container.mk to use native docker compose from Docker Desktop
- improve development docs

### Checklist
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: docs/development/DEVELOPMENT.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
 # Developing the package
 
-Prerequisites
+## Prerequisites
 
-* (Optional but highly recommended) [pipx](https://pypa.github.io/pipx/).
-* [Nox](https://nox.thea.codes/) for all automation.
+* Python 3.9
```
Comment (tatiana) [inline docs/development/DEVELOPMENT.md:5]: We support Python 3.7, 3.8 and 3.9 ATM

Comment (feluelle) [inline docs/development/DEVELOPMENT.md:5]:
```suggestion
* At least Python 3.7, 3.8 or 3.9
```

Comment (feluelle) [reply docs/development/DEVELOPMENT.md:5]: WDYT @tatiana?

Comment (kaxil) [reply docs/development/DEVELOPMENT.md:5]: Fine for me

File: docs/development/DEVELOPMENT.md [hunk 2]
```diff
@@ -140,50 +151,50 @@ You can configure the Docker-based testing environment to test your DAG
...
-2. Run ``make container target=build-run``
+1. Run ``make container target=build-run``
 
-3. Put the DAGs you want to run in the dev/dags directory:
+1. Put the DAGs you want to run in the dev/dags directory:
 
-4. If you want to add Connections, create a connections.yaml file in the dev directory.
+1. If you want to add Connections, create a connections.yaml file in the dev directory.
 
    See the `Connections Guide <https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html>`_ for more information.
```
Comment (kaxil) [inline docs/development/DEVELOPMENT.md:160]: Not related to this PR, but this link is RST not MD, we should fix that too

File: mk/container.mk [hunk 3]
```diff
@@ -3,17 +3,17 @@ PHONY: build-run clean logs stop help
 .DEFAULT_GOAL:= help
 
 logs: ## View logs of the all the containers
-	docker-compose -f ../dev/docker-compose.yaml logs --follow
+	docker compose -f ../dev/docker-compose.yaml logs --follow
```
Comment (kaxil) [inline mk/container.mk:6]: Why this change?

Comment (kaxil) [reply mk/container.mk:6]: TIL: Looks like "docker compose" is the new way approach and written in GO as compared to `docker-composer`. 

https://docs.docker.com/compose/#compose-v2-and-the-new-docker-compose-command

cc @astronomer/airflow-providers - We should update it in astronomer-providers repo too :)

**Review by kaxil — APPROVED** [review]

File: mk/container.mk (hunk 3 above)
Comment (feluelle) [reply mk/container.mk:6]: Yes, exactly. It is much nicer also in terms of output. I believe the v1 docker-compose did not have coloured output and is not collapsible.
